File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/SolidCheckList.java
package ch.bailu.aat_lib.preferences;


import javax.annotation.Nonnull;

public abstract class SolidCheckList extends AbsSolidType {

    public abstract String[] getStringArray();
    public abstract boolean[] getEnabledArray();
    public abstract void setEnabled(int i, boolean isChecked);




    @Override
    public void setValueFromString(String string) {}

    @Nonnull
    @Override
    public String getValueAsString() {
        return "";
    }

}

File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/SolidCheckList.kt
package ch.bailu.aat_lib.preferences

abstract class SolidCheckList : AbsSolidType() {

    abstract fun getStringArray(): Array<String>
    abstract fun getEnabledArray(): BooleanArray

    abstract fun setEnabled(index: Int, isChecked: Boolean)
    override fun setValueFromString(string: String) {}

    override fun getValueAsString(): String {
        return ""
    }
}
